### PR TITLE
GH-48972: [Python] Add errors='coerce' parameter to pyarrow.compute.cast

### PR DESCRIPTION
**Pull request by shashbha14**
### Rationale for this change
Currently, `pyarrow.compute.cast()` raises an exception when encountering values that cannot be cast to the target type. This PR adds an `errors` parameter (similar to `pandas.to_numeric(errors='coerce')`) to allow gracefully handling cast failures by converting them to null values.
### What changes are included in this PR?
- Added `errors` parameter to `cast()` function with options `'raise'` (default) and `'coerce'`
- When `errors='coerce'`, sets `CastOptions.null_on_error = True` to produce nulls for invalid casts
- Added `is_castable()` meta-function to check element-wise castability
- Comprehensive test suite in `test_coerce_cast.py`
### Are these changes tested?
Yes, added extensive tests covering:
- Basic coercion behavior
- Edge cases (empty strings, "nan", None values)
- Temporal type casting
- Integration with existing `safe` parameter
- Instance methods (`array.cast()`, `chunked_array.cast()`)
- Error validation
### Are there any user-facing changes?
Yes - new `errors` parameter for `pyarrow.compute.cast()` and new `is_castable()` function.

* GitHub Issue: #48972

### CODE REVIEW

File: dev/release/download_rc_binaries.py
Comment (raulcd) [inline dev/release/download_rc_binaries.py:139]: the commit updating this shouldn't be part of this PR, right?

File: python/pyarrow/table.pxi [hunk 1]
```diff
@@ -2105,6 +2116,17 @@ cdef class _Tabular(_PandasConvertible):
...
+    def personal_data(self):
+        """
+        Whether the object contains personal data.
+
+        Returns
+        -------
+        personal_data : bool
+        """
+        return self.schema.personal_data
+
```
Comment (raulcd) [inline python/pyarrow/table.pxi:2129]: this seems completely unrelated.

**Review by raulcd — CHANGES_REQUESTED** [review]
Thanks for the PR. Apart from the failures on linting and CI. There are other things on this PR that shouldn't be part of it. Could you try and remove the unrelated changes?

**Comment by rok** [comment]
Closing for lack of activity.

**Comment by rok** [comment]
@shashbha14 please feel free to reopen if you want to continue the work.
